src/repository.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for handling interactions with git repository."""
+
+import base64
+import logging
+import re
+from pathlib import Path
+
+from git import GitCommandError
+from git.repo import Repo
+from github import Github
+from github.GithubException import GithubException, UnknownObjectException
+from github.Repository import Repository
+
+from .constants import DOCUMENTATION_FOLDER_NAME
+from .exceptions import InputError, RepositoryClientError, RepositoryFileNotFoundError
+
+GITHUB_HOSTNAME = "github.com"
+ORIGIN_NAME = "origin"
+HTTPS_URL_PATTERN = re.compile(rf"^https?:\/\/.*@?{GITHUB_HOSTNAME}\/(.+\/.+?)(.git)?$")
```

Comment:
What if the remote url is using ssh protocol? For example `git@github.com:canonical/abc.git`.

Reply:
By default the github actions uses http protocol, so i think it is safe to parse the HTTPS only

Reply:
We should consider this for https://github.com/canonical/upload-charm-docs/issues/18, up to now since the action runs in GitHub actions only `https` should be fine for now